homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,188 @@
+                        "stimeout": "5000000",
+                    }
+                else:
+                    stream_options = {}
+                if info.get(CONF_RTSP_TRANSPORT):
+                    stream_options[FFMPEG_OPTION_MAP[CONF_RTSP_TRANSPORT]] = info[
+                        CONF_RTSP_TRANSPORT
+                    ]
+                container = await self.hass.async_add_executor_job(
+                    partial(
+                        av.open,
+                        info[CONF_STREAM_SOURCE],
+                        options=stream_options,
+                        timeout=SOURCE_TIMEOUT,
+                    )
+                )
+                video_stream = container.streams.video[0]
+                if video_stream is not None:
+                    return True, ""
+            except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+                return False, "stream_unauthorised"
+            except (KeyError, IndexError):
+                return False, "stream_novideo"
+            except PermissionError:
+                return False, "stream_not_permitted"
+            except OSError as err:
+                if "No route to host" in str(err):
+                    return False, "stream_no_route_to_host"
+                if "Input/output error" in str(err):
+                    return False, "stream_io_error"
+                raise err
+        return True, ""
+
+    @staticmethod
+    def build_schema(user_input):
+        """Create schema for camera config setup."""
+        spec = {
+            vol.Optional(CONF_NAME, default=user_input[CONF_NAME]): str,
+            vol.Required(
+                CONF_STILL_IMAGE_URL,
+                default=user_input.get(CONF_STILL_IMAGE_URL),
+            ): str,
+            vol.Optional(
+                CONF_STREAM_SOURCE,
+                default=user_input.get(CONF_STREAM_SOURCE, ""),
+            ): str,
+            vol.Optional(
+                CONF_RTSP_TRANSPORT, default=user_input.get(CONF_RTSP_TRANSPORT, None)
+            ): vol.In([None, "tcp", "udp", "udp_multicast", "http"]),
+            vol.Optional(
+                CONF_AUTHENTICATION, default=user_input[CONF_AUTHENTICATION]
+            ): vol.In([HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]),
+            vol.Optional(
+                CONF_USERNAME,
+                default=user_input.get(CONF_USERNAME, ""),
+            ): str,
+            vol.Optional(CONF_PASSWORD, default=user_input.get(CONF_PASSWORD, "")): str,
+            vol.Optional(
+                CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+                default=user_input[CONF_LIMIT_REFETCH_TO_URL_CHANGE],
+            ): bool,
+            vol.Optional(CONF_CONTENT_TYPE, default=user_input[CONF_CONTENT_TYPE]): str,
+            vol.Optional(CONF_FRAMERATE, default=user_input[CONF_FRAMERATE]): int,
+            vol.Optional(CONF_VERIFY_SSL, default=user_input[CONF_VERIFY_SSL]): bool,
```

Comment:
If these are optional parameters, do you need to use `user_input.get` instead for the defaults?

Reply:
Agreed. Done.